### ISSUE TITLE
Why should the filters equal to 3 * (classes + 5) ?

### ISSUE THREAD

**Issue by ClarrkLau** [AlexeyAB/darknet#4164]
I need to train yolov3-tiny to detect my own object. And I learned that the filters should be changed to 3 * (classes + 5) in `conv` layer before each `yolo` layer, but why? Is it mentioned in the paper, if so where is it? And where could I learn more about this? Thanks in advance.

**Comment by kossolax** [comment]
3, for the number of used anchor in this layer
5, for 1 objectness and 4 bounding box offsets

**Comment by AlexeyAB** [comment]
https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects

> (Generally filters depends on the classes, coords and number of masks, i.e. filters=(classes + coords + 1)*<number of mask>, where mask is indices of anchors. If mask is absence, then filters=(classes + coords + 1)*num)

**Comment by ClarrkLau** [comment]
@AlexeyAB @kossolax Thank you very much !

I still have some questions, here is the thing:
I'm a beginner in yolo and deep learning. Recently I'm trying to train yolo for detecting scuba diver, but account some problems about how to train.
1. I am trying to train yolov3-tiny, used a weights download from https://pjreddie.com/darknet/yolo/, and partial it to generate a pre-trained weights use following command,
`./darknet partial cfg/yolov3-tiny.cfg yolov3-tiny.weights yolov3-tiny.conv.15 15`
but this pre-trained weights is for a 80-classes model, while I just need the model to predict only one class, is it ok just modify the classes parameter equal to 1 in yolo layer of the .cfg file? Wouldn't that decrease the performance? Is there any other modification should I do?
2. If I consider scuba diver as a person class in COCO, what pre-trained weights should I use for training. The original yolov3-tiny.weights or a partial one? 
3. In the above two methods, how large the dataset should I prepare for my custom object? 

Great thanks in advance!
